### ISSUE TITLE
YAML Validator Breaks Jinja-Templated YAML Prompt

### ISSUE THREAD

**Issue by jay0129** [promptfoo/promptfoo#1225]
There was an update validating the YAML format, and my previous test cases failed. 

error:
```
/opt/homebrew/lib/node_modules/promptfoo/node_modules/js-yaml/lib/loader.js:183
  return new YAMLException(message, mark);
         ^
YAMLException: end of the stream or a document separator is expected (7:1)

 4 |     Respond in {{ language }}.
 5 |     If the Human message is an e ...
 6 | 
 7 | {% for message in previous_messa ...
-----^
```

prompt:
```
- role: system
  content: Blah

{% for message in previous_messages %}
  {% for role, content in message %}
- role: "{{ role }}"
  content: {{ content | dump }}
  {% endfor %}
{% endfor %}

- role: user
  content: {{ user_input }}
```

By the way, I am not sure what happens to the JSON format? 
https://github.com/promptfoo/promptfoo/blob/main/examples/openai-chat-history/prompt.json

**Comment by typpo** [comment]
Thanks for flagging, should be fixed in #1226.  I'll wait for @mldangelo to have a look before merging.

**Comment by mldangelo** [comment]
@jay0129 we just deployed a fix for this in [0.73.1](https://github.com/promptfoo/promptfoo/releases/tag/0.73.1). Thank you for flagging it!
